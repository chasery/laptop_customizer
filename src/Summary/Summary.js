import React from "react";
import SummaryOption from "../SummaryOption/SummaryOption";
import SummaryTotal from "../SummaryTotal/SummaryTotal";

class Summary extends React.Component {
  render() {
    const summary = Object.keys(this.props.selected).map((feature, idx) => {
      const featureHash = feature + "-" + idx;
      const selectedOption = this.props.selected[feature];

      return (
        <SummaryOption
          key={featureHash}
          feature={feature}
          selectedOptionName={selectedOption.name}
          selectedOptionCost={selectedOption.cost}
        />
      );
    });

    return (
      <section className="main__summary">
        <h2>Your cart</h2>
        {summary}
        <SummaryTotal selected={this.props.selected} />
      </section>
    );
  }
}

Summary.defaultProps = {
  features: [],
  selected: [],
};

export default Summary;
