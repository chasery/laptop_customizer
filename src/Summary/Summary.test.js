import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Summary from "./Summary";

describe("Summary component", () => {
  it("renders without errors", () => {
    const wrapper = shallow(<Summary />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
