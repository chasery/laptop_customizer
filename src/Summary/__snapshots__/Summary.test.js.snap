// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary component renders without errors 1`] = `
<section
  className="main__summary"
>
  <h2>
    Your cart
  </h2>
  <SummaryTotal
    selected={Array []}
  />
</section>
`;
